### ISSUE TITLE
undefined function xdebug_disable()

### ISSUE THREAD

**Issue by xdosil** [markshust/docker-magento#528]
**Description**
I'm creating the environment for an existing project, but, when I try the step "bin/magento app:config:import" show a fatal error with this missage: "Fatal error: Uncaught Error: Call to undefined function xdebug_disable() in /var/www/html/vendor/magento/magento2-functional-testing-framework/src/Magento/FunctionalTestingFramework/_bootstrap.php:73"

**Steps To Reproduce**
1. Download the Docker Compose template:
2. Replace with existing source code of your existing Magento instance:
3. Create a DNS host entry for the site:
4. Start some containers, copy files to them and then restart the containers:
5. Import existing database:
bin/mysql < backups/magento.sql
6. Import app-specific environment settings:
bin/magento app:config:import

**Expected Result**
I don't know

**Actual Result**
Fatal error: Uncaught Error: Call to undefined function xdebug_disable() in /var/www/html/vendor/magento/magento2-functional-testing-framework/src/Magento/FunctionalTestingFramework/_bootstrap.php:73

**Comment by markshust** [comment]
Is anyone else able to replicate this? It appears to be working for me:

![Screen Shot 2021-09-07 at 9 58 43 AM](https://user-images.githubusercontent.com/437029/132357651-94067a1f-1ee0-4435-a73f-009a2450078c.png)

**Comment by xdosil** [comment]
Thanks Mark.

I'm not sure about the problem, but I found this: https://magento.stackexchange.com/questions/327971/uncaught-error-call-to-undefined-function-xdebug-disable

After read the post, I installed the xdebug 2.9.8 version, and removed the `magento2-functional-testing-framework` package but the problem still showing.

I don't understand.

Thanks!

**Comment by kenbradleywhite** [comment]
@markshust I've replicated this

On 7.3 I was getting issues with Imagick, so I tried upgrading to `image: markoshust/magento-php:7.4-fpm`

On `bin/restart`  I got

Status: Downloaded newer image for markoshust/magento-php:7.4-fpm
Starting kegland-magento-docker_rabbitmq_1      ... done
Starting kegland-magento-docker_db_1            ... done
Starting kegland-magento-docker_redis_1         ... done
Starting kegland-magento-docker_mailhog_1       ... done
Starting kegland-magento-docker_elasticsearch_1 ... done
Recreating kegland-magento-docker_phpfpm_1      ... done
Recreating kegland-magento-docker_app_1         ... done

$ 20:36:43 ERROR: failed shelling out to php for reading the module list.
20:36:43 ERROR: failed shelling out to php for reading the theme list.
^C
```

Then when running `setup:upgrade`:

```
$ bin/bash
app@4a676505244d:~/html$ bin/magento set:up

Fatal error: Uncaught Error: Call to undefined function xdebug_disable() in /var/www/html/vendor/magento/magento2-functional-testing-framework/src/Magento/FunctionalTestingFramework/_bootstrap.php:73
Stack trace:
#0 /var/www/html/vendor/composer/autoload_real.php(70): require()
#1 /var/www/html/vendor/composer/autoload_real.php(60): composerRequire74a20b85c60a49806a204f70602c3b57('eda65932675b68b...', '/var/www/html/v...')
#2 /var/www/html/vendor/autoload.php(7): ComposerAutoloaderInit74a20b85c60a49806a204f70602c3b57::getLoader()
#3 /var/www/html/app/autoload.php(30): include('/var/www/html/v...')
#4 /var/www/html/app/bootstrap.php(33): require_once('/var/www/html/a...')
#5 /var/www/html/bin/magento(14): require('/var/www/html/a...')
#6 {main}
  thrown in /var/www/html/vendor/magento/magento2-functional-testing-framework/src/Magento/FunctionalTestingFramework/_bootstrap.php on line 73

```

**Comment by kenbradleywhite** [comment]
I've switched my docker-compose.yml to `php:7.4-fpm-1` and I no longer have the issue.

So the issue is with the `php:7.4`-fpm image

**Comment by kenbradleywhite** [comment]
@markshust I've replicated this

On 7.3 I was getting issues with Imagick, so I tried upgrading to `image: markoshust/magento-php:7.4-fpm`

On `bin/restart`  I got

```
Status: Downloaded newer image for markoshust/magento-php:7.4-fpm
Starting magento-docker_rabbitmq_1      ... done
Starting magento-docker_db_1            ... done
Starting magento-docker_redis_1         ... done
Starting magento-docker_mailhog_1       ... done
Starting magento-docker_elasticsearch_1 ... done
Recreating magento-docker_phpfpm_1      ... done
Recreating magento-docker_app_1         ... done

$ 20:36:43 ERROR: failed shelling out to php for reading the module list.
20:36:43 ERROR: failed shelling out to php for reading the theme list.
^C
```

Then when running `setup:upgrade`:

```
$ bin/bash
app@4a676505244d:~/html$ bin/magento set:up

Fatal error: Uncaught Error: Call to undefined function xdebug_disable() in /var/www/html/vendor/magento/magento2-functional-testing-framework/src/Magento/FunctionalTestingFramework/_bootstrap.php:73
Stack trace:
#0 /var/www/html/vendor/composer/autoload_real.php(70): require()
#1 /var/www/html/vendor/composer/autoload_real.php(60): composerRequire74a20b85c60a49806a204f70602c3b57('eda65932675b68b...', '/var/www/html/v...')
#2 /var/www/html/vendor/autoload.php(7): ComposerAutoloaderInit74a20b85c60a49806a204f70602c3b57::getLoader()
#3 /var/www/html/app/autoload.php(30): include('/var/www/html/v...')
#4 /var/www/html/app/bootstrap.php(33): require_once('/var/www/html/a...')
#5 /var/www/html/bin/magento(14): require('/var/www/html/a...')
#6 {main}
  thrown in /var/www/html/vendor/magento/magento2-functional-testing-framework/src/Magento/FunctionalTestingFramework/_bootstrap.php on line 73

```

**Comment by kenbradleywhite** [comment]
I've switched the image from `php:7.4-fpm` to `php:7.4-fpm-1` and no longer have the issue

So the issue must be with `php:7.4-fpm`

**Comment by kenbradleywhite** [comment]
The same applies to 7.3. `php:7.3-fpm` has the issue, `php:7.3-fpm-1` doesn't

**Comment by piotrkwiecinski** [comment]
@kenbradleywhite The issue is related to an old version of `magento/magento2-functional-testing-framework` it was fixed in version 3.2.1 https://github.com/magento/magento2-functional-testing-framework/issues/808

If you want to use xdebug 3 you have you use a new version of the package or downgrade to an older image with xdebug 2 as you did above.

**Comment by markshust** [comment]
Thanks @piotrkwiecinski -- going to close out this issue, as it looks like there is a code workaround and isn't specific to this Docker setup.

As far as `php:7.3-fpm` vs `php:7.3-fpm-1`, those aren't related at all. The `-1` is a very old image, while the base tag `php:7.3-fpm` references the most recent tag, `php:7.3-fpm-14`:

![Screen Shot 2021-09-17 at 3 04 30 PM](https://user-images.githubusercontent.com/437029/133840761-b31e415d-7e3e-4b53-9a2d-82ea792c1d48.png)
